### PR TITLE
implement intensity compensation for VC-1 decoding for Ivybridge and better

### PR DESCRIPTION
**Pull request by carpalis**
VC1-decoder: implement intensity compensation for VC-1

Intensity compensation was not present for B-frames, but only for P-frames. When a P-frame flags intensity compensation for its forward reference frame, all subsequent B-frames that use this reference frame as well, need to do intensity compensation as well.

Fixes: #13 [IVB/BYT/BSW] Macroblocking artifact seen on a VC1 video file

Signed-off-by: Jerome Borsboom <jerome.borsboom@carpalis.nl>

### CODE REVIEW

**Comment by xhaihao** [comment]
Could you please add a line of 'Fixes #xxx' to the commit message so that others know which issue is resolved by this commit?

File: src/gen75_mfd.c [hunk 1]
```diff
@@ -1915,24 +1934,37 @@ gen75_mfd_vc1_pred_pipe_state(VADriverContextP ctx,
...
+    picture_type = pic_param->picture_fields.bits.picture_type;
+
+    if (gen7_mfd_context->reference_surface[0].surface_id != VA_INVALID_ID) {
+        if (picture_type == 1 || picture_type == 2) { /* P/B picture */
+            struct gen7_vc1_surface *gen7_vc1_surface = gen7_mfd_context->reference_surface[0].obj_surface->private_data;
+
+            intensitycomp_single_fwd = gen7_vc1_surface->intensity_compensation;
+            luma_scale1 = gen7_vc1_surface->luma_scale;
+            luma_shift1 = gen7_vc1_surface->luma_shift;
+        }
```
Comment (xhaihao) [inline src/gen75_mfd.c]: Does P frame also use the parameters from the reference frame? the commit message only says all subsequent B-frames need to do intensity compensation.

Comment (carpalis) [inline src/gen75_mfd.c]: The commit message only referenced the changes in intensity compensation. For P-frames intensity compensation was already working.

It is the P-frame that sets the parameters for its reference frame. Subsequently, the P-frame also uses this intensity compensated reference frame. Any B-frames that follow this P-frame and that reference that same reference frame also use the intensity compensation on the reference frame. (See 8.3.8 in VC-1 spec).

The patch stores the intensity compensation status with each surface. When such a surface is used as a reference, the stored intensity compensation status us used to program the graphics hardware.

Comment (xhaihao) [inline src/gen75_mfd.c]: thanks for the explanation in detail

**Pull request merged**